zerver/lib/types.py

```diff
@@ -260,6 +260,23 @@ class APISubscriptionDict(APIStreamDict):
     subscribers: List[int]
 
 
+@dataclass
+class Hotspot:
+    name: str
+    title: Optional[StrPromise]
+    description: Optional[StrPromise]
+    has_trigger: bool = False
+
+    def to_dict(self, delay: float = 0) -> Dict[str, Union[str, float, bool]]:
+        return {
+            "name": self.name,
+            "title": str(self.title),
+            "description": str(self.description),
+            "delay": delay,
+            "has_trigger": self.has_trigger,
+        }
```

Comment:
Let's leave this type definition at the top of `hotspots.py`; I think it'll be a better experience for most code that needs to interact with this to do so.